### DISCUSSION TITLE
E2SM Ran Control Slice-level PRB Quota messages should skip the number 4

### DISCUSSION THREAD

**Discussion by gckopper** [srsran/srsRAN_Project#713] [General Help] [answered]
Hello everyone!

In the issue #468 was discussed that the messages sent by the example xApp provided with flexric were not being decoded correctly by srsran, in part, because they didn't skip the number 4 when numbering the parameter ids, as per specification. Since then, the br-flexric branch of flexric has fixed the issue with [this commit](https://gitlab.eurecom.fr/mosaic5g/flexric/-/commit/a79610c31d3d1075997532718b9e23134c113c1c). However, srsran also changed its numbering so it doesn't skip the number 4. Now  srsran's decoding is not in spec nor is the encoding in the xapp provided by the [srsran/oran-sc-ric](https://github.com/srsran/oran-sc-ric/blob/main/xApps/python/lib/e2sm_rc_module.py) repo. 

I'd like to know if this change was made intentionally in order to conform with the older version of flexric or if this is a bug? 

Here is a screenshot from the spec (O-RAN.WG3.E2SM-RC-R003-v06.00) for quick reference.
![image](https://github.com/srsran/srsRAN_Project/assets/41166074/de84cebc-7c29-446b-874f-bd781c3bad3a)

**Comment by pgawlowicz** [comment, marked as answer]
Hi, thanks for pointing this out, we will match the order from the spec in the next release.

**Reply by yagoda** [reply]
@gckopper,
Indeed we had set it intentionally to be interoperable with FlexRic, now that they have changed this we will do so too.

**Comment by yagoda** [comment]
Hi @gckopper,

In preparation for our upcoming workshop in DC, we are contacting some users of our E2 solution. Can you please share a few lines about your experience with the E2 interface, what you were trying to achieve, and how it worked out for you?

Kind Regards,
Justin
